doc/varnish-configuration.rst

```diff
@@ -36,111 +36,162 @@ will be used throughout the Varnish examples on this page.
 Provided Vcl Subroutines
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-In order to ease configuration we provide a set of vcl subroutines in the resources/config directory.
-These can be directly included into ``your_varnish.vcl`` and the needed subroutines called
-from the respective vcl_* subroutines.
+In order to ease configuration we provide a set of vcl subroutines in the
+``resources/config`` directory. These can be included from your main Varnish
+configuration file, typically ``default.vcl``. Then you need to make your
+``vcl_*`` subroutines call the ``fos_*`` routines.
 
-.. important::
-    When including one of the provided vcl you need to call all the defined subroutines
-    or your configuration will not be valid.
+.. tip::
+    When including one of the provided vcl, you need to call all the defined
+    subroutines or your configuration will not be valid.
 
-    See the respective sections below on how to configure usage of each of the provided vcl's.
+    See the respective sections below on how to configure usage of each of the
+    provided vcl's.
 
 Purge
 ~~~~~
 
-To configure Varnish for `handling PURGE requests <https://www.varnish-cache.org/docs/3.0/tutorial/purging.html>`_:
-
-Purge removes a specific URL (including query strings) in all its variants (as specified by the ``Vary`` header).
+Purge removes a specific URL (including query strings) in all its variants (as
+specified by the ``Vary`` header).
 
 Subroutines are provided in ``resources/config/varnish-[version]/fos_purge.vcl``.
-
 To enable support add the following to ``your_varnish.vcl``:
 
-.. code-block:: varnish3
+.. configuration-block::
 
-    include "path-to-config/varnish-[version]/fos_purge.vcl";
+    .. code-block:: varnish4
 
-.. configuration-block::
-    .. literalinclude:: ../tests/Functional/Fixtures/varnish-4/fos.vcl
-        :language: varnish4
-        :lines: 17,19,21-22
-    .. literalinclude:: ../tests/Functional/Fixtures/varnish-3/fos.vcl
-        :language: varnish3
-        :lines: 15,17,19-20,25-32
+        include "path-to-config/varnish-4/fos_purge.vcl";
+
+        sub vcl_recv {
+            call fos_purge_recv;
+        }
```

Comment:
:+1: on writing out VCL instead of the includes. Let’s keep the includes for larger VCL sections only.